opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -642,26 +644,20 @@ def __init__(
         self._at_exit_handler = None
         if shutdown_on_exit:
             self._at_exit_handler = atexit.register(self.shutdown)
+        self._logger_cache = {}
+        self._logger_cache_lock = Lock()
 
     @property
     def resource(self):
         return self._resource
 
-    def get_logger(
+    def _get_logger_no_cache(
```

Comment:
are we missing a test for this?

Reply:
This test checks for correctness of get_logger with attributes https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/tests/logs/test_logs.py#L48

Should I add a no_cache case to the cache test or benchmark as well? I initially left it out as attributes aren't accessible through the standard logger but happy to add it!

Reply:
I would expect a simple no_cache case in the cache test that shows when using get_logger() passing attributes there's no entry in _logger_cache. Does it make sense to you?

Reply:
makes sense, I added 2 cases to directly test the 2 code paths (with cache and without)